1145 email minor interest

hoerstl opened the pull request:
## Issue
Have an "Email All Interested" button to allow the admin to send a message to each student who has expressed interest in the CCE Minor. Resolves #1145

## Solution
Make a button that emails everyone who's inside the interested students table. 
We did remove the ability to show how many people are in that table to put the button there (see image). 
### Before
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/e9396328-e3d1-46a3-a74d-bec0ecc3b216)
### After
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/9ff03949-1d52-4e77-8a3c-77633f95af1e)
Let us know if this is a problem and we'll figure out something that keeps the functionality and is cosmetically awesome.

## To test
- Navigate to the minor management page under Admin in the sidebar.
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/655dee80-bbff-471d-9570-0bb48e3a4318)
- Click the email all button by the interested students table and ensure that a window opens with your system's default email application
- Ensure that the email addresses from the table match the users in the table.

hoerstl committed:
Started porting the email modal to the minor page

hoerstl committed:
Added an email all button using mailto, aborting the plan to use the email modal

hoerstl commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Have an "Email All Interested" button on the CCE Minor Management page](https://trello.com/c/WupWMMtP/735-have-an-email-all-interested-button-on-the-cce-minor-management-page)

azabeli reviewed (dismissed):
This looks good to me and it works perfect. The only suggestion I would give is to maybe add some message as a sample email , if there is one.

bledsoef committed:
Merge branch 'development' into 1145-Email-Minor-Interest

bledsoef commented:
I think it looks good overall. Can I ask why you removed the ability to filter how many people are shown in the table, though?

bledsoef reviewed (dismissed):
Personally, I would not remove the page filtering in the top left and try to find a better place for the button. But if you guys feel that is the best way to do it I will approve it.

andersoncedu commented:
> Personally, I would not remove the page filtering in the top left and try to find a better place for the button. But if you guys feel that is the best way to do it I will approve it.

This was going to be the same comment I made. Also, the mailto isn't working for me, it's not opening an email in outlook,

hoerstl committed:
Moved the button and added an info message

hoerstl committed:
Merge branch '1145-Email-Minor-Interest' of github.com:BCStudentSoftwareDevTeam/celts into 1145-Email-Minor-Interest

hoerstl committed:
Merge branch 'development' of github.com:BCStudentSoftwareDevTeam/celts into 1145-Email-Minor-Interest

hoerstl commented:
These are good concerns, I have moved the button. How do ya like it?
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/53ff6af5-3a06-4f40-b3a2-8f9eec561ed2)

stevensonmichel commented on app/templates/admin/cceMinor.html:
```diff
@@ -45,21 +45,31 @@ <h2>CCE Minor Progress</h2>
     <tbody>
+        {% set interestedStudentEmails = [] %}
         {% for student in interestedStudentsList %}
+            {% set _ = interestedStudentEmails.append(student.email) %}
```
If there is a possibility to handle that in the backend before rendering, you should go for it. Having this operation done in the route would be awesome, given that jinja should be the last option for this kind of operation.

hoerstl replied:
Good idea. This has been done

stevensonmichel commented:
After clicking on Email All, the emails of the interested students are inproperly added, giving this impression of a whole long email.

[
![Screenshot 2024-04-24 101231](https://github.com/BCStudentSoftwareDevTeam/celts/assets/112141243/66ce969c-f158-49da-8905-5a1f6524fa89)
](url)

hoerstl committed:
Fixed a bug where student emails were not being separated in the mailto link

hoerstl commented:
> After clicking on Email All, the emails of the interested students are inproperly added, giving this impression of a whole long email.
> 
> [ ![Screenshot 2024-04-24 101231](https://private-user-images.githubusercontent.com/112141243/325257349-66ce969c-f158-49da-8905-5a1f6524fa89.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQwNjk2NDEsIm5iZiI6MTcxNDA2OTM0MSwicGF0aCI6Ii8xMTIxNDEyNDMvMzI1MjU3MzQ5LTY2Y2U5NjljLWYxNTgtNDlkYS04OTA1LTVhMWY2NTI0ZmE4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyNVQxODIyMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NzEwZWY1ZGI3YmU3NTM3MWVlNDhjYzk1YWE4M2RhZmU3MDkxMTdhNDRlNzQzYjliNGFlMWZkMjk1MGVjOWNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.H85Z3EOWiboEFigY-nfc7PyCBJxPy36OHgea5WwvBl4) ](url)

Thanks! This issue is fixed now.

andersoncedu commented:
You have tests failing for your PR.

hoerstl committed:
Fixed a type annotation

AndersonStettner reviewed (commented):
The only two things I have for now are the "X" on the flash message that I talked with Lawrence about and the `Email All` button. 

Due to how close the tables are the button, to me, looks like it could be for the CCE Minor Progress table or the Interested Students table. The suggestion I have for this is adding a bit of space between the two tables so they are a bit more separated. This is just a suggestions though, if yall are fine with it and others are then it is fine.

hoerstl committed:
Added a close button to msgFlash and refactored how flashed messages were being created from the backend.

hoerstl committed:
Added some space between the tables on the minor page

hoerstl commented:
> The only two things I have for now are the "X" on the flash message that I talked with Lawrence about and the `Email All` button.
> 
> Due to how close the tables are the button, to me, looks like it could be for the CCE Minor Progress table or the Interested Students table. The suggestion I have for this is adding a bit of space between the two tables so they are a bit more separated. This is just a suggestions though, if yall are fine with it and others are then it is fine.

Comment addressed. I've both added a close button for sure onto the flashed messages and refactored how messages were being displayed from the backend to avoid redundant code. Additionally, I added some space between the two tables as per your suggestion which I liked.
